### ISSUE TITLE
Python 3 compatibility with dictionary views vs lists.

### ISSUE THREAD

**Issue by rjkoch** [diffpy/diffpy.srfit#60]
So, if I try to run a co-refinement, where I have more than one fit contribution, generally the recipe building goes fine.

But, I if call something like `FitResults(recipe).printResults()` I get:

```
...
AttributeError: 'odict_keys' object has no attribute 'sort'

```

Something similar occurs if I call `recipe.setWeights`

I think this is because dictionary keys/values etc are being treated as lists (python 2)

I'll see if I can fix it.

**Comment by sbillinge** [comment]
Thanks Rob.

It looks as if it might be an ordered dict, is it?  You can't sort a dict,
but you may be able to sort an ordered dict.  I think those are in package
containers.  Anyway, look into it, we may be able to get the same behavior
this way, though I am just guessing.

**Comment by rjkoch** [comment]
The issue is that '.sort()' and '.index()' are being called on the return
of 'dict.keys()' and 'dict.values()'

In python 2, 'keys()' and 'values()' return lists, so  '.sort()' and
'.index()' are fine. In python 3,  'keys()' and 'values()' do not return
lists, so this breaks. The simplest solution I think is to wrap these
'dict.keys()' and 'dict.values()' calls in 'list()'

I've already fixed the two cases I encountered in a cloned local repo. I
just need to figure out how to submit a pull request (it's will be my very
first).

The more difficult part will be implementing this in method/function
returns that are then used by other functions. It's not clear to me when
the return should be a list, a view, or a generator.

Rob
